vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/PollItem.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2020 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.home.room.detail.timeline.item
+
+import android.widget.LinearLayout
+import android.widget.TextView
+import com.airbnb.epoxy.EpoxyAttribute
+import com.airbnb.epoxy.EpoxyModelClass
+import im.vector.app.R
+import im.vector.app.features.home.room.detail.RoomDetailAction
+import im.vector.app.features.home.room.detail.timeline.TimelineEventController
+import org.matrix.android.sdk.api.extensions.orFalse
+import org.matrix.android.sdk.api.session.room.model.message.MessagePollContent
+
+@EpoxyModelClass(layout = R.layout.item_timeline_event_base)
+abstract class PollItem : AbsMessageItem<PollItem.Holder>() {
+
+    @EpoxyAttribute
+    var pollContent: MessagePollContent? = null
+
+    @EpoxyAttribute
+    var pollResponseSummary: PollResponseData? = null
+
+    @EpoxyAttribute
+    var callback: TimelineEventController.Callback? = null
+
+    @EpoxyAttribute
+    var eventId: String? = null
+
+    @EpoxyAttribute
+    var pollSent: Boolean = false
+
+    override fun bind(holder: Holder) {
+        super.bind(holder)
+        val relatedEventId = eventId ?: return
+
+        renderSendState(holder.view, holder.questionTextView)
+
+        holder.questionTextView.text = pollContent?.pollCreationInfo?.question?.question
+
+        holder.optionsContainer.removeAllViews()
```

Comment:
Should be better to avoid to do that each time, but rather get the existing views and recycle them.

Reply:
Done. We always need to remove & add views but I suggest a way to prevent the recreation of option views.